tests/system_tests_link_routes.py

```diff
@@ -1797,110 +1814,227 @@ def _fake_broker(self, cls):
         self.router.wait_connectors()
         return fake_broker
 
+    def _find_frame(self, data, code) -> Optional[list]:
```

Comment:
This could be typed more. What is the type of `data` is quite obvious. I have little idea what is the type of the `code` parameter (I suspect `int`) and I am not sure list of _what_ is being returned. I'd venture a guess it is an `int` as well.

```suggestion
    def _find_frame(self, data: Data, code: int) -> Optional[List[int]]:
```